src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
+  const reorderedDate = [];
+  const oldYear = newDate[oldYearIndex];
+
+  function getIndex(index) {
+    return index.indexOf('YY') !== -1
+      ? index.indexOf('YY')
+      : index.indexOf('YYYY');
+  }
+
+  function getYear(number, from, to) {
+    if (from > to) {
+      return number.slice(2);
+    }
+
+    if (from < to) {
+      return number < 30 ? '20' + number : '19' + number;
+    }
+
+    return number;
+  }
+
+  newDate[oldYearIndex] = getYear(oldYear, oldYearLength, newYearLength);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (i === newYearIndex) {
+      reorderedDate[i] = newDate[oldYearIndex];
+    } else {
+      const index = fromFormat.indexOf(toFormat[i]);
+
+      reorderedDate[i] = newDate[index];
+    }
+  }
```

Comment:
In the for loop, you are using the indexOf method inside the loop. This is inefficient as it results in a time complexity of O(n^2). Instead, you can create a map of the fromFormat array outside the loop and then use it inside the loop to get the index in O(1) time.